rust-lib/src/lib.rs

```diff
@@ -20,14 +22,19 @@ use rgb::i9n::*;
 use rgb::rgbd::ContractName;
 use rgb::util::SealSpec;
 
+#[macro_use]
+extern crate amplify_derive;
+
 trait CReturnType: Sized + 'static {
-    fn from_opaque(other: &COpaqueStruct) -> Result<&mut Self, String> {
+    fn from_opaque(other: &COpaqueStruct) -> Result<&mut Self, RequestError> {
         let mut hasher = DefaultHasher::new();
         TypeId::of::<Self>().hash(&mut hasher);
         let ty = hasher.finish();
 
         if other.ty != ty {
-            return Err(String::from("Type mismatch"));
+            return Err(RequestError::Runtime(
+                rgb::error::BootstrapError::ArgParseError("Type mismatch".to_string()),
```

Comment:
is it ok to create the error as a `BootstrapError::ArgParseError`?

Reply:
I think its ok

Reply:
Btw, for the future, we have `s!(..)` macro for easier string conversion so you can do `ArgParseError(s!("Type mismatch"))`

Reply:
cool, next time I'll use that, thanks for the hint